package com.cl.service.impl;

import org.springframework.stereotype.Service;
import java.util.Map;
import java.util.List;

import com.baomidou.mybatisplus.mapper.Wrapper;
import com.baomidou.mybatisplus.mapper.EntityWrapper;
import com.baomidou.mybatisplus.plugins.Page;
import com.baomidou.mybatisplus.service.impl.ServiceImpl;
import com.cl.utils.PageUtils;
import com.cl.utils.Query;


import com.cl.dao.JiangxuejinshenqingDao;
import com.cl.entity.JiangxuejinshenqingEntity;
import com.cl.service.JiangxuejinshenqingService;
import com.cl.entity.view.JiangxuejinshenqingView;

@Service("jiangxuejinshenqingService")
public class JiangxuejinshenqingServiceImpl extends ServiceImpl<JiangxuejinshenqingDao, JiangxuejinshenqingEntity> implements JiangxuejinshenqingService {

    	
    @Override
    public PageUtils queryPage(Map<String, Object> params) {
        Page<JiangxuejinshenqingEntity> page = this.selectPage(
                new Query<JiangxuejinshenqingEntity>(params).getPage(),
                new EntityWrapper<JiangxuejinshenqingEntity>()
        );
        return new PageUtils(page);
    }
    
    @Override
	public PageUtils queryPage(Map<String, Object> params, Wrapper<JiangxuejinshenqingEntity> wrapper) {
		  Page<JiangxuejinshenqingView> page =new Query<JiangxuejinshenqingView>(params).getPage();
	        page.setRecords(baseMapper.selectListView(page,wrapper));
	    	PageUtils pageUtil = new PageUtils(page);
	    	return pageUtil;
 	}
    
	@Override
	public List<JiangxuejinshenqingView> selectListView(Wrapper<JiangxuejinshenqingEntity> wrapper) {
		return baseMapper.selectListView(wrapper);
	}

	@Override
	public JiangxuejinshenqingView selectView(Wrapper<JiangxuejinshenqingEntity> wrapper) {
		return baseMapper.selectView(wrapper);
	}
	
	

    @Override
    public List<Map<String, Object>> selectValue(Map<String, Object> params, Wrapper<JiangxuejinshenqingEntity> wrapper) {
        return baseMapper.selectValue(params, wrapper);
    }

    @Override
    public List<Map<String, Object>> selectTimeStatValue(Map<String, Object> params, Wrapper<JiangxuejinshenqingEntity> wrapper) {
        return baseMapper.selectTimeStatValue(params, wrapper);
    }

    @Override
    public List<Map<String, Object>> selectGroup(Map<String, Object> params, Wrapper<JiangxuejinshenqingEntity> wrapper) {
        return baseMapper.selectGroup(params, wrapper);
    }




}
